rasa/nlu/components.py

```diff
@@ -650,10 +663,22 @@ def prepare_partial_processing(
         self.partial_processing_pipeline = pipeline
         self.partial_processing_context = context
 
+    def prepare_partial_training(self, training_data: TrainingData) -> None:
+        """Prepare the component for training on just a part of the data.
+
+        The complete training data can be used to extract some information
+        that might be missing when just a part of the training data is
+        seen during training.
+
+        Args:
+            training_data: The complete training data.
+        """
+        pass
+
     def partially_process(self, message: Message) -> Message:
```

Comment:
Optional: Can we change the name to avoid the name overlap with the existing `process` method?
On that note, what's the intended use of this method? I am confused between this and `prepare_partial_training`

Reply:
I am not 100% sure. The docstring says:
```
Allows the component to process messages during training (e.g. external training data).

The passed message will be processed by all components previous to this one in the pipeline.
```
I also cannot find any usages of this method, so we might be able to remove this method.